.eslintrc.js

```diff
@@ -1,19 +1,78 @@
-"use strict";
-
 module.exports = {
-  root: true,
-  extends: [
-    "eslint:recommended",
-    "plugin:eslint-plugin/recommended",
-    "plugin:node/recommended",
-  ],
+  parser: '@typescript-eslint/parser',
+  parserOptions: {
+    project: './tsconfig.json',
+    ecmaVersion: 6,
+    sourceType: "module",
+    ecmaFeatures: {
+      modules: true
+    }
+  },
   env: {
     node: true,
   },
-  overrides: [
-    {
-      files: ["tests/**/*.js"],
-      env: { mocha: true },
-    },
+  settings: {
+    'import/resolver': {
+      'node': {
+        'extensions': ['.js', '.ts'],
+      },
+      'typescript': {
+        "alwaysTryTypes": true,
```

Comment:
What is this for? Can you add a comment?

Reply:
It used for libraries that contains only types definitions. For instance it is `@types/estree` library.
I have added the comment in code.